specs/agents/configuration.md

```diff
@@ -143,3 +143,48 @@ By default, agents MUST send data to the APM Server at `http://localhost:8200/`.
 If possible, agents SHOULD detect sensible defaults for `service.name` and `service.version`.
 In any case agents MUST include `service.name` - if discovering it is not possible then the default value: `unknown-${service.agent.name}-service` MUST be used.
 This naming pattern allows the UI to display inline help on how to manually configure the name.
+
+### Adding a new configuration option
+
+When adding a new configuration option to the spec, please use the following template:
+
+#### <option_name_in_central_config_notation> configuration
+
+The description of the option.
+Try to phrase it in a way that lets agents easily copy/paste the description into their configuration documentation page.
+This should also match the description in the [central configuration](https://github.com/elastic/kibana/blob/main/x-pack/plugins/apm/common/agent_configuration/setting_definitions/general_settings.ts) in the APM UI.
+
+|                |                                                                         |
+|----------------|-------------------------------------------------------------------------|
+| Type           | See the [Configuration Value Types](#configuration-value-types) section |
+| Default        | `<default value>`                                                       |
+| Central config | `<true/false>`                                                          |
+
+Additional description of the option that should not be part of the documentation.
+For example, whether agents SHOULD or MUST implement the option.
+
+#### Adding to central configuration
+
+One of the things to determine when proposing to add a new configuration option is whether it should be added to central configuration.
+In general, we should try to make the majority of options available in central configuration.
+However, sometimes it might not be feasible if agents need to read the value at startup and changing the value at runtime is very difficult to achieve.
+
+When adding a centrally configurable option,
+the spec owner is expected to create a pull request in the Kibana repository to add the option to central configuration.
+As all the configuration options are added declaratively, this is no rocket science.
+See this example PR on how to add an option: https://github.com/elastic/kibana/pull/84678.
+
+Agent devs are not expected to set up a local Kibana development environment.
+Just create a draft PR using your best judgement given the examples on how to add the configuration option.
+By the time you're adding the configuration option, chances are that no agent has implemented the option, yet.
+Therefore, set `includeAgents: []` in the option declaration.
+When an agent implements the option, they just need to add the agent name to this section, without having to test the changes.
```

Comment:
For example, when I did https://github.com/elastic/kibana/pull/85655/files way back, there *was* a test file change to make.

Reply:
I've changed this in https://github.com/elastic/kibana/pull/143668 so that the tests don't need to be updated.